sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamSetOperatorRelBase.java

```diff
@@ -88,15 +92,20 @@ public BeamSetOperatorRelBase(BeamRelNode beamRelNode, OpType opType, boolean al
                 leftRows.apply(
                     "CreateLeftIndex",
                     MapElements.via(new BeamSetOperatorsTransforms.BeamSqlRow2KvFn())))
-            .and(
-                rightTag,
-                rightRows.apply(
-                    "CreateRightIndex",
-                    MapElements.via(new BeamSetOperatorsTransforms.BeamSqlRow2KvFn())))
-            .apply(CoGroupByKey.create());
+                .and(
```

Comment:
note that to make it general, it would be a for loop to create one tag per input.